src/main/java/dal/fatih/todoly/HibernateTaskRepository.java

```diff
@@ -0,0 +1,112 @@
+package dal.fatih.todoly;
+
+
+import javax.persistence.*;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.persistence.metamodel.EntityType;
+import java.sql.Date;
+import java.time.Instant;
+import java.util.List;
+import java.util.Locale;
+
+public class HibernateTaskRepository implements TaskRepository {
+
+    private final EntityManagerFactory entityManagerFactory
+            = Persistence.createEntityManagerFactory("Todoly");
+    private EntityManager entityManager = entityManagerFactory.createEntityManager();
+    private CriteriaBuilder cb;
+    private CriteriaQuery<Task> cq;
+    private Root<Task> taskRoot;
+
+
+    public void getBaseQuery() {
+        cb = entityManager.getCriteriaBuilder();
+        cq = cb.createQuery(Task.class);
+        taskRoot = cq.from(Task.class);
+    }
+
+    @Override
+    public boolean create(Task task) {
+        EntityTransaction et = null;
+        try {
+            et = entityManager.getTransaction();
+            et.begin();
+            entityManager.persist(task);
+            et.commit();
+
+            return true;
+        } catch (Exception e) {
+            if (et != null) {
+                et.rollback();
+                return false;
+            }
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public List<Task> list() {
+        getBaseQuery();

```

Comment:
Same here. This is a horrible practice. Never ever do such a thing. Always have your methods returning something and assign it to a newly created variable.